www/profile/index.php

```diff
@@ -0,0 +1,171 @@
+<?php
+
+$servername = "localhost";
+$username = "worker";
+$password = "swiffcat";
+$db = "socialmedia";
+
+// Create connection
+$conn = new mysqli($servername, $username, $password, $db);
+
+// Check connection
+if (!$conn) {
+    die("Connection failed: " . mysqli_connect_error());
+}
+
+$headers = getallheaders();
+if (!array_key_exists("token", $headers)) {
+    header("HTTP/1.1 401 Unauthorized");
+    die("{\"reason\":\"token not included\"}");
+}
+$token = getallheaders()["token"];
+
+// If invalid header token, die
+$stmt = $conn->prepare("SELECT user_id FROM auth WHERE token = ?");
+$stmt->bind_param("s", $a1);
+$a1 = $token;
+$stmt->execute();
+$stmt->store_result();
+$user_id = null;
+$stmt->bind_result($user_id);
+if ($stmt->fetch()) {
+    // user_id is bound
+} else {
+    header("HTTP/1.1 401 Unauthorized");
+    die();
+}
+
+if ($_SERVER['REQUEST_METHOD'] === 'GET') {
+    if (!isset($_GET["id"])) {
+        header("HTTP/1.1 403 Forbidden");
+        die("{\"reason\":\"no user id\"}");
+    }
+
+    $user_id = $_GET["id"];
+
+    // make sure the user exists
+    $stmt = $conn->prepare("SELECT user_id FROM users WHERE user_id = ?");
+    $stmt->bind_param("s", $a1);
+    $a1 = $user_id;
+    $stmt->execute();
+    $stmt->store_result();
+    $count = $stmt->num_rows;
+    if ($count != 1) {
+        header("HTTP/1.1 403 Forbidden");
+        die("{\"reason\":\"invalid user\"}");
+    }
+
+    // Retrieve data from the posts table
+    $stmt = $conn->prepare("SELECT * FROM posts WHERE user_id = ? ORDER BY created_at DESC");
+    $stmt->bind_param("s", $a1);
+    $a1 = $user_id;
+    $stmt->execute();
+    $result = $stmt->get_result();
+    $posts = array();
+
+    if ($result->num_rows > 0) {
+        while ($row = $result->fetch_assoc()) {
+            $posts[] = $row;
+        }
+    }
+
+    // Retrieve User profile data from users table
+    $stmt = $conn->prepare("SELECT * FROM users WHERE user_id = ?");
+    $stmt->bind_param("s", $a1);
+    $a1 = $user_id;
+    $stmt->execute();
+    $result = $stmt->get_result();
+    $row = $result->fetch_assoc();
+    // add posts to the values
+    $row["posts"] = $posts;
+    $json = json_encode($row);
+    echo $json;
+
+    // Close the database connection
+    mysqli_close($conn);
+}
+
+if ($_SERVER['REQUEST_METHOD'] === 'POST') {
```

Comment:
Why would posting to "profile" be what this is doing? You don't think a different endpoint makes more sense?

Reply:
I suppose I could post to user instead, but it made sense to me that if the user was updating data for their profile page it would be a post to the profile for updating it.